src/wmtk/operations/tet_mesh/TetSplitWithTags.hpp

```diff
@@ -19,8 +19,10 @@ struct OperationSettings<tet_mesh::TetSplitWithTags>
     MeshAttributeHandle<double> pos_handle;
     long split_vertex_tag_value;
 
+    OperationSettings(const Mesh& m): invariants(m) {}
+
     InvariantCollection invariants;
-    void initialize_invariants(const TetMesh& m);
+    void initialize_invariants();
```

Comment:
initializing invariants no longer takes an arg as it should read the mesh() from the invariantcollection's mesh held internally

Reply:
if you use shared_ptr then just create the invariantcollection using make_Shared and then add things as we currently are doing